File: decompiled/sources/com/hjq/toast/ActivityStack.java
package com.hjq.toast;

import android.app.Activity;
import android.app.Application;
import android.os.Bundle;

final class ActivityStack implements Application.ActivityLifecycleCallbacks {
    private Activity mForegroundActivity;

    public void onActivityCreated(Activity activity, Bundle bundle) {
    }

    public void onActivityDestroyed(Activity activity) {
    }

    public void onActivitySaveInstanceState(Activity activity, Bundle bundle) {
    }

    public void onActivityStarted(Activity activity) {
    }

    public void onActivityStopped(Activity activity) {
    }

    ActivityStack() {
    }

    static ActivityStack register(Application application) {
        ActivityStack activityStack = new ActivityStack();
        application.registerActivityLifecycleCallbacks(activityStack);
        return activityStack;
    }

    public Activity getForegroundActivity() {
        return this.mForegroundActivity;
    }

    public void onActivityResumed(Activity activity) {
        this.mForegroundActivity = activity;
    }

    public void onActivityPaused(Activity activity) {
        if (this.mForegroundActivity == activity) {
            this.mForegroundActivity = null;
        }
    }
}

File: decompiled/sources/com/hjq/toast/ActivityStack.kt
package com.hjq.toast

import android.app.Activity
import android.app.Application
import android.os.Bundle

class ActivityStack private constructor() : Application.ActivityLifecycleCallbacks {
    private var mForegroundActivity: Activity? = null

    override fun onActivityCreated(activity: Activity, bundle: Bundle?) {
    }

    override fun onActivityDestroyed(activity: Activity) {
    }

    override fun onActivitySaveInstanceState(activity: Activity, bundle: Bundle) {
    }

    override fun onActivityStarted(activity: Activity) {
    }

    override fun onActivityStopped(activity: Activity) {
    }

    fun getForegroundActivity(): Activity? = mForegroundActivity

    override fun onActivityResumed(activity: Activity) {
        mForegroundActivity = activity
    }

    override fun onActivityPaused(activity: Activity) {
        if (mForegroundActivity === activity) {
            mForegroundActivity = null
        }
    }

    companion object {
        @JvmStatic
        fun register(application: Application): ActivityStack {
            val activityStack = ActivityStack()
            application.registerActivityLifecycleCallbacks(activityStack)
            return activityStack
        }
    }
}
